src/Illuminate/Support/Collection.php

```diff
@@ -726,17 +726,20 @@ public function forPage($page, $perPage)
     }
 
     /**
-     * Partition the collection into two array using the given callback.
+     * Partition the collection into two arrays using the given callback or key.
      *
-     * @param  callable  $callback
+     * @param  callable|string  $callback
+     * @param  bool  $preserveKeys
      * @return array
      */
-    public function partition(callable $callback)
+    public function partition($callback, $preserveKeys = false)
```

Comment:
Collections operations generally should preserve keys by default and there should be no option not to as you can call array_values or ->values() if you don't want to.